spring-integration-core/src/main/java/org/springframework/integration/splitter/AbstractMessageSplitter.java

```diff
@@ -116,6 +123,26 @@ else if (result instanceof Iterator<?>) {
 		return builder;
 	}
 
+	/**
+	 * Return true if the subclass needs to add headers in the resulting splits.
+	 * If true, {@link #addHeaders(Message, AbstractIntegrationMessageBuilder)} will be called.
```

Comment:
There is no such a method already, because you have changed it just to the `Map`.
But we are forced to come back to the `AbstractIntegrationMessageBuilder` because `Map`:
```
  * @since 1.8
     */
    default V putIfAbsent(K key, V value) {
```